src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -215,102 +345,118 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
+            LOGGER.log(Level.INFO, "Determining analysis id...");
+            analysisId = workspace.act(new DetermineAid(expandedCodesonarProjectFile));
+            LOGGER.log(Level.INFO, "Found analysis id: {0}", analysisId);
         } else {
-            LOGGER.log(Level.INFO, "[CodeSonar] Using override analysis id: '" + aid + "'.");
+            analysisId = aid;
+            LOGGER.log(Level.INFO, "Using override analysis id: \"" + aid + "\".");
         }
         
-        String analysisUrl = baseHubUri.toString() + "/analysis/" + aid + ".xml";
+        String analysisUrl = baseHubUri.toString() + "/analysis/" + analysisId + ".xml";
 
         Analysis analysisWarnings = analysisService.getAnalysisFromUrlWarningsByFilter(analysisUrl);
-        URI metricsUri = metricsService.getMetricsUriFromAnAnalysisId(baseHubUri, aid);
+        URI metricsUri = metricsService.getMetricsUriFromAnAnalysisId(baseHubUri, analysisId);
         Metrics metrics = metricsService.getMetricsFromUri(metricsUri);
-        URI proceduresUri = proceduresService.getProceduresUriFromAnAnalysisId(baseHubUri, aid);
+        URI proceduresUri = proceduresService.getProceduresUriFromAnAnalysisId(baseHubUri, analysisId);
         Procedures procedures = proceduresService.getProceduresFromUri(proceduresUri);
 
         Analysis analysisNewWarnings = analysisService.getAnalysisFromUrlWithNewWarnings(analysisUrl);
         List<Pair<String, String>> conditionNamesAndResults = new ArrayList<>();
-
-        CodeSonarBuildActionDTO buildActionDTO = new CodeSonarBuildActionDTO(analysisWarnings, analysisNewWarnings, metrics, procedures, baseHubUri);
+        
+        Long lAnalysisId = null;
+        try {
+            lAnalysisId = Long.valueOf(analysisId);
+        } catch(NumberFormatException e) {
+            LOGGER.log(Level.WARNING, "Unable to parse analysis id \"" + analysisId + "\" as long integer.");
+        }
+        CodeSonarBuildActionDTO buildActionDTO = new CodeSonarBuildActionDTO(lAnalysisId, analysisWarnings, analysisNewWarnings, metrics, procedures, baseHubUri);
         CodeSonarBuildAction csba = new CodeSonarBuildAction(buildActionDTO, run, expandedProjectName, analysisUrl);
         
-        listener.getLogger().println("[CodeSonar] Finding previous builds for comparison");
+        csLogger.writeInfo("Finding previous builds for comparison");
         
         CodeSonarBuildActionDTO compareDTO = null;
         Run<?,?> previosSuccess = run.getPreviousSuccessfulBuild();
         if(previosSuccess != null) {
-            listener.getLogger().println("[CodeSonar] Found previous build to compare to");
+            csLogger.writeInfo("Found previous build to compare to ({0})", previosSuccess.getDisplayName());
             List<CodeSonarBuildAction> actions = previosSuccess.getActions(CodeSonarBuildAction.class).stream().filter(c -> c.getProjectName() != null && c.getProjectName().equals(expandedProjectName)).collect(Collectors.toList());
```

Comment:
Can we break this line of code into multiple, shorter lines?  It is over 200 chars long and it does some important work that should not be overlooked.